### PR TITLE
An alternative OCI Object Storage source

### PR DESCRIPTION
**Pull request by amir**
Uses AbstractTamerJob instead of AbstractStatefulSourceTamerJob

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -13,15 +13,17 @@ lazy val V = new {
...
   val postgres      = "42.2.19"
   val refined       = "0.9.21"
   val scalacheck    = "1.15.3"
   val scalatest     = "3.2.6"
   val silencer      = "1.7.3"
   val zio           = "1.0.5"
-  val `zio-s3`      = "0.3.0"
   val `zio-interop` = "2.3.1.0"
   val `zio-kafka`   = "0.14.0"
+  val `zio-oci-os`  = "0.1.2"
```
Comment (amir) [inline build.sbt:25]: pending release

**Comment by sirocchj** [comment]
I personally think this is much cleaner than #275

**Review by barambani — APPROVED** [review]

File: oci-objectstorage/src/main/scala/tamer/oci/objectstorage/OciObjectStorageConfiguration.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  def objectName(state: S): Option[String]
+}
+
+object OciObjectStorageConfiguration {
+  case class State[R, K, V, S](
+      initialState: S,
+      getNextState: (S, Option[String]) => URIO[R, S],
+      deriveKafkaRecordKey: (S, V) => K
+  )
+}
```
Comment (sirocchj) [inline oci-objectstorage/src/main/scala/tamer/oci/objectstorage/OciObjectStorageConfiguration.scala:21]: this should probably move back in common as the "core" configuration all frontends need, kind of merged back with `SourceConfiguration`. Not for now, but would love to see s3 follow and then some cleanup

**Review by sirocchj — APPROVED** [review]

**Comment by sirocchj** [comment]
@amir If you're happy too we could merge this one in and close #275 

I'd like to get it tested at least as S3 is but there's no testkit/standalone docker for OCI's Object Storage as far as I can tell.. So we'd probably need to resort to mocking... Also, we could do it as a followup.

I'd be more interested to use this to align the S3 implementation and cleanup some of the abstractions we have introduced in core and see what that would look like.

**Comment by amir** [comment]
@sirocchj agreed on testing, I'll merge this one now and work on testing apparatus.

**Comment by amir** [comment]
@sirocchj I don't seem to be have the permission to merge this as snyk is failing. Do you mind merging it?

**Pull request merged**